src/053._maximum_subarray.md

```diff
@@ -0,0 +1,44 @@
+## Maximum Subarray
+
+### Description
+
+```Python
+Given an integer array nums, find the contiguous subarray (containing at least one number) which has the largest sum and return its sum.
+
+Example:
+    Input: [-2,1,-3,4,-1,2,1,-5,4],
+    Output: 6
+    Explanation: [4,-1,2,1] has the largest sum = 6.
+
+Follow up:
+
+If you have figured out the O(n) solution, try coding another solution using the divide and conquer approach, which is more subtle.
+```
+
+### Python Solutions
+
+#### Approach : Dynamic Programming
+
+* Time complexity : O(n)
+* Space complexity : O(n)
+
+```Python
+class Solution(object):
+    def maxSubArray(self, nums):
+        """
+        Given an integer array nums, find the contiguous subarray (containing at least one number) which has the largest sum and return its sum
+
+        Args:
+            nums(List[int]): the name of array of integers
+        
+        Returns:
+            the largest contiguous subarray sum(int)
+        """
+        n = len(nums)
+        if n == 1:
+            return nums[0]
+        maxSum = [nums[0] for i in range(n)]
+        for i in range(1, n):
+            maxSum[i] = max(maxSum[i - 1] + nums[i], nums[i])
+        return max(maxSum)
+```
```

Comment:
It is possible to reduce space complexity to O(1). No need to create a new array.